api/repository/pokemons.repository.ts

```diff
@@ -0,0 +1,25 @@
+import { Pokemon } from "../interfaces/pokemons.interfaces";
+import { prisma } from "../utils/prisma";
+import { Prisma } from "@prisma/client";
+
+export const pokemonRepository = {
+  getRandomPokemon,
+};
+
+async function getRandomPokemon(listedIds: string[]) {
+  const query =
+    listedIds.length > 0
+      ? Prisma.sql`
+    SELECT * FROM "Pokemon"
+    WHERE id NOT IN (${Prisma.join(listedIds)})
+    ORDER BY RANDOM()
+    LIMIT 1
```

Comment:
```suggestion
async function getRandomPokemon(listedIds: string[]) {
const filter =  listedIds.length > 0
      ? Prisma.sql`WHERE id NOT IN (${Prisma.join(listedIds)})`
      : '';
  const query = Prisma.sql`
    SELECT * FROM "Pokemon"
    ${filter}
    ORDER BY RANDOM()
    LIMIT 1
```

Reply:
The problem is that Prisma only allows certain things, like lists with `Prisma.join`, to be injected into `Prisma.sql`—you can't directly insert conditions or other SQL fragments like that.